src/membership.rs

```diff
@@ -0,0 +1,381 @@
+    /// Remember to remove this code in next release, or guard the code with the spec version
+    // Given it is easy to forget to do this, we need a better arrangement.
+    fn runtime_initialization(spec_version: u32) {
+        if 5 == spec_version {
+            print("Running New Runtime Init Code");
+            let default_terms: PaidMembershipTerms<T> = Default::default();
+            <PaidMembershipTermsById<T>>::insert(T::PaidTermId::sa(0), default_terms);
+        }
+    }
+}
+
+decl_module! {
+    pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+        fn deposit_event<T>() = default;
+
+        fn on_initialise(_now: T::BlockNumber) {
+            if Self::store_spec_version().map_or(true, |spec_version| VERSION.spec_version > spec_version) {
+                // Mark store version with latest so we don't run initialization code again
+                <StoreSpecVersion<T>>::put(VERSION.spec_version);
+                Self::runtime_initialization(VERSION.spec_version);
+            }
+        }
+
+        fn on_finalise(_now: T::BlockNumber) {
+
+        }
+
+        /// Non-members can buy membership
+        fn buy_membership(origin, paid_terms_id: T::PaidTermId, user_info: UserInfo) {
+            let who = ensure_signed(origin)?;
+
+            // make sure we are accepting new memberships
+            ensure!(Self::new_memberships_allowed(), "new members not allowed");
+
+            // ensure key not associated with an existing membership
+            Self::ensure_not_member(&who)?;
+
+            // ensure paid_terms_id is active
+            let terms = Self::ensure_active_terms_id(paid_terms_id)?;
+
+            // ensure enough free balance to cover terms fees
+            ensure!(T::Currency::can_slash(&who, terms.fee), "not enough balance to buy membership");
+
+            let user_info = Self::check_user_registration_info(user_info)?;
+
+            // ensure handle is not already registered
+            Self::ensure_unique_handle(&user_info.handle)?;
+
+            let _ = T::Currency::slash(&who, terms.fee);
+            let member_id = Self::insert_new_paid_member(&who, paid_terms_id, &user_info);
+
+            Self::deposit_event(RawEvent::MemberRegistered(member_id, who.clone()));
+        }
+
+        fn change_member_about_text(origin, text: Vec<u8>) {
+            let who = ensure_signed(origin)?;
+            Self::_change_member_about_text(&who, &text)?;
+        }
+
+        fn change_member_avatar(origin, uri: Vec<u8>) {
+            let who = ensure_signed(origin)?;
+            Self::_change_member_avatar(&who, &uri)?;
+        }
+
+        /// Change member's handle.
+        fn change_member_handle(origin, handle: Vec<u8>) {
+            let who = ensure_signed(origin)?;
+            Self::_change_member_handle(&who, handle)?;
+        }
+
+        fn batch_change_member_profile(origin, user_info: UserInfo) {
```

Comment:
maybe just `update_profile`?
```suggestion
        fn update_profile(origin, user_info: UserInfo) {
```